Cabal/Distribution/Simple/GHC.hs

```diff
@@ -577,7 +577,11 @@ buildOrReplLib mReplFlags verbosity numJobs pkg_descr lbi lib clbi = do
       linkerOpts = mempty {
                       ghcOptLinkOptions       = PD.ldOptions libBi
```

Comment:
What's the difference between `PD.ldOptions libBi` and options for `ldProgram` ? How their origin differ?

Reply:
* `PD.ldOptions libBi` contains the `ld-options` of the `library` section in the `.cabal` file.
  * The corresponding `PD.ldOptions bnfo` further down for when the executable is built contains the `ld-options` of the `executable` section in the `.cabal` file.
* The `programOverrideArgs` for `ldProgram` is what's given with the `--ld-options` CLI flag to e.g. `cabal build` or `cabal configure`.
  * Interestingly they are duplicated 2x, that is, if you give `--ld-options=-s` then `maybe [] programOverrideArgs (lookupProgram ldProgram (withPrograms lbi))` is `["-s","-s"]`.

Reply:
> I haven't figured out yet why that is, as that's a separate issue.

I have figured that out now. It happens for the new `ld-options` just as much as for the already-existing `gcc-options`. I've found the reason for that and file #6183 for it.